boards/posix/native_posix/Kconfig

```diff
@@ -4,6 +4,7 @@ config BOARD_NATIVE_POSIX
 	bool
 	select NATIVE_POSIX_TIMER
 	select POSIX_ARCH_CONSOLE
+	select NATIVE_APPLICATION
```

Comment:
technically not needed as `native_posix` board sets the `SOC_POSIX=y` which selects `ARCH_POSIX`, which cause `NATIVE_APPLICATION` to default to `y`. (applies also to `nrf52_bsim` board)

But does no harm either.

Reply:
That is correct, but it seemed more correct to have it select'ed from the board. 
The default was provided to not break out of tree boards.